packages/protocol/contracts/bridge/libs/LibBridgeProcess.sol

```diff
@@ -76,30 +74,36 @@ library LibBridgeProcess {
             "B:notReceived"
         );
 
-        // We retrieve the necessary ether from EtherVault
+        // We retrieve the necessary ether from EtherVault if receiving on
+        // Taiko, otherwise it is already available in this Bridge.
         address ethVault = resolver.resolve("ether_vault", true);
         if (ethVault != address(0)) {
             EtherVault(payable(ethVault)).releaseEther(
                 message.depositValue + message.callValue + message.processingFee
             );
         }
-        // We deposit Ether first before the message call in case the call
-        // will actually consume the Ether.
+        // We send the Ether before the message call in case the call will
+        // actually consume Ether.
         message.owner.sendEther(message.depositValue);
 
         LibBridgeStatus.MessageStatus status;
         uint256 refundAmount;
 
+        // TODO(dave): can we just catch this earlier (in sendMessage) and throw an error?
```

Comment:
This is good suggestion. I created [this PR](https://github.com/taikoxyz/taiko-mono/pull/13107/files) to check message.to on the source chain, but I think since the bridges on the source chain and the destination chain are separately deployed and may have different version, may be we should keep the checks here as well.

Reply:
that is fine with me, approved the PR